### PR TITLE
Convert Stats & Clock to functional components

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves 

This converts the Stats & Clock components of the NTP to functional components. As an added bonus, it uses the new pref hooks instead of prop drilling.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_new_tab_ui/hooks/PrefHookManager.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export class PrefHookManager<T> {
         getInitialValue().then(this.notifyListeners)
     }
 
-    notifyListeners (prefs: T) {
+    notifyListeners = (prefs: T) => {
```
Comment (petemill) [inline components/brave_new_tab_ui/hooks/PrefHookManager.ts:30]: Just wondering why this change is neccessary?

File: components/brave_new_tab_ui/reducers/new_tab_reducer.ts [hunk 2]
```diff
@@ -204,15 +204,6 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
...
 
-    case types.UPDATE_CLOCK_WIDGET: {
-      const { showClockWidget, clockFormat } = payload
-      performSideEffect(async function (state) {
-        preferencesAPI.saveShowClock(showClockWidget)
-        preferencesAPI.saveClockFormat(clockFormat)
-      })
-      break
-    }
-
```
Comment (petemill) [inline components/brave_new_tab_ui/reducers/new_tab_reducer.ts:215]: 🎉

File: components/brave_new_tab_ui/components/default/clock/index.tsx [hunk 3]
```diff
@@ -3,100 +3,46 @@
...
+  // Don't render AM/PM
+  const formattedTime = React.useMemo(() => formatter.formatToParts(now)
+    .map(t => t.type === 'dayPeriod'
+      ? null
+      : t.value), [formatter, now])
+
+  return <StyledClock onDoubleClick={toggleClockFormat}>
+    <StyledTime>{formattedTime}</StyledTime>
+  </StyledClock>
 }
```
Comment (petemill) [inline components/brave_new_tab_ui/components/default/clock/index.tsx:48]: so much nicer to read!

**Review by petemill — APPROVED** [review]
Really nice and readable - thanks!

File: components/brave_new_tab_ui/hooks/PrefHookManager.ts (hunk 1 above)
Comment (fallaciousreasoning) [reply components/brave_new_tab_ui/hooks/PrefHookManager.ts:30]: I think there was a case where the this context was being set to something else

**Comment by bsclifton** [comment]
Nice cleanup @fallaciousreasoning 😄
